src/org/gk/model/InstanceUtilities.java

```diff
@@ -1394,4 +1396,62 @@ else if (instance.getSchemClass().isa(ReactomeJavaConstants.Pathway))
 	//public static String encodeLineSeparators(String text) {
 	//    return text.replaceAll(FileAdaptor.LINE_END + "", "<br>");
 	//}
+
+     /**
+      *
+      * @param id
+      * @return
+      */
+     public static boolean isDrug(Long id) {
+         if (id == null)
+             return false;
+
+         XMLFileAdaptor fileAdaptor = PersistenceManager.getManager().getActiveFileAdaptor();
+         return isDrug(fileAdaptor.fetchInstance(id));
+     }
+
+     /**
+      * Determines if an instance is a drug or contains a drug (in the case of EntitySets and Complexes).
+      *
+      * @param instance
+      * @throws Exception
+      * @returns boolean
+      */
+     public static boolean isDrug(Object instance) {
+         if (instance == null)
+             return false;
+
+         SchemaClass schemaClass = ((GKInstance) instance).getSchemClass();
+         // Check if instance is a drug.
+         if (schemaClass.isa(ReactomeJavaConstants.Drug))
+             return true;
+
+         List<String> validClasses = Arrays.asList(ReactomeJavaConstants.EntitySet,
+                                                   ReactomeJavaConstants.Complex);
+         // Check id instance is an EntitySet or Complex.
+         if (validClasses.stream().noneMatch(schemaClass::isa))
+             return false;
+
+         // If EntitySet has a "hasMember" or "hasCandidate" attribute then recursively iterate over them.
+         Set<GKInstance> containedInstances = new HashSet<GKInstance>();
+         try {
+             containedInstances = getContainedInstances((GKInstance) instance,
+                                                        ReactomeJavaConstants.hasMember,
+                                                        ReactomeJavaConstants.hasCandidate,
+                                                        ReactomeJavaConstants.hasComponent);
+         } catch (Exception e) {
+             e.printStackTrace();
+         }
+         // No containedInstances.
+         if (containedInstances.size() == 0 || containedInstances == null)
+             return false;
```

Comment:
Try to throw the exception in the method. Otherwise, you will return false in case there is any exception in the above try, which may not be what you want.

Reply:
Fixed in [ad58739](https://github.com/reactome/CuratorTool/pull/26/commits/3822611dfb0486ef3eb45d492b83cc810ad58739).